### PR TITLE
Reduce scope of skipping tests on device without fp64 aspect

### PR DESCRIPTION
**Pull request by antonwolfy**
Since #1515 is closed, the tests relating to `dpnp.reminder` and `dpnp.mod` can be unmuted, since they are not failing anymore on a device without fp64 aspect.
The TR also removed skipped tests for another already elementwise function leveraged from dpctl, such as `dpnp.add`, `dpnp.divide`, `dpnp.subtract` and some others.

- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [ ] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: tests/skipped_tests_gpu_no_fp64.tbl [hunk 1]
```diff
@@ -930,1155 +930,249 @@ tests/third_party/cupy/manipulation_tests/test_basic.py::TestCopytoFromScalar_pa
...
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_137_{arg1=array([-3., -2., -1.,  1.,  2.,  3.], dtype=float32), arg2=array([-3., -2., -1.,  1.,  2.,  3.]), dtype=float64, name='true_divide', use_dtype=False}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_138_{arg1=array([-3., -2., -1.,  1.,  2.,  3.], dtype=float32), arg2=array([-3., -2., -1.,  1.,  2.,  3.]), dtype=float64, name='subtract', use_dtype=True}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_139_{arg1=array([-3., -2., -1.,  1.,  2.,  3.], dtype=float32), arg2=array([-3., -2., -1.,  1.,  2.,  3.]), dtype=float64, name='subtract', use_dtype=False}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_140_{arg1=array([-3., -2., -1.,  1.,  2.,  3.], dtype=float32), arg2=0.0, dtype=float64, name='power', use_dtype=True}::test_binary
```
Comment (vlad-perevezentsev) [inline tests/skipped_tests_gpu_no_fp64.tbl:1236]: This test falls on Iris Xe and should not have been removed

```
tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_140_{arg1=array([-3., -2., -1.,  1.,  2.,  3.], dtype=float32), arg2=0.0, dtype=float64, name='power', use_dtype=True}::test_binary FAILED [  8%]
```

File: tests/skipped_tests_gpu_no_fp64.tbl [hunk 2]
```diff
@@ -930,1155 +930,249 @@ tests/third_party/cupy/manipulation_tests/test_basic.py::TestCopytoFromScalar_pa
...
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_161_{arg1=array([-3., -2., -1.,  1.,  2.,  3.]), arg2=array([-3., -2., -1.,  1.,  2.,  3.], dtype=float32), dtype=float64, name='true_divide', use_dtype=False}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_162_{arg1=array([-3., -2., -1.,  1.,  2.,  3.]), arg2=array([-3., -2., -1.,  1.,  2.,  3.], dtype=float32), dtype=float64, name='subtract', use_dtype=True}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_163_{arg1=array([-3., -2., -1.,  1.,  2.,  3.]), arg2=array([-3., -2., -1.,  1.,  2.,  3.], dtype=float32), dtype=float64, name='subtract', use_dtype=False}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_164_{arg1=array([-3., -2., -1.,  1.,  2.,  3.]), arg2=array([-3., -2., -1.,  1.,  2.,  3.]), dtype=float64, name='power', use_dtype=True}::test_binary
```
Comment (vlad-perevezentsev) [inline tests/skipped_tests_gpu_no_fp64.tbl:1251]: The same here

File: tests/skipped_tests_gpu_no_fp64.tbl [hunk 3]
```diff
@@ -930,1155 +930,249 @@ tests/third_party/cupy/manipulation_tests/test_basic.py::TestCopytoFromScalar_pa
...
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_166_{arg1=array([-3., -2., -1.,  1.,  2.,  3.]), arg2=array([-3., -2., -1.,  1.,  2.,  3.]), dtype=float64, name='true_divide', use_dtype=True}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_167_{arg1=array([-3., -2., -1.,  1.,  2.,  3.]), arg2=array([-3., -2., -1.,  1.,  2.,  3.]), dtype=float64, name='true_divide', use_dtype=False}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_168_{arg1=array([-3., -2., -1.,  1.,  2.,  3.]), arg2=array([-3., -2., -1.,  1.,  2.,  3.]), dtype=float64, name='subtract', use_dtype=True}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_169_{arg1=array([-3., -2., -1.,  1.,  2.,  3.]), arg2=array([-3., -2., -1.,  1.,  2.,  3.]), dtype=float64, name='subtract', use_dtype=False}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_170_{arg1=array([-3., -2., -1.,  1.,  2.,  3.]), arg2=0.0, dtype=float64, name='power', use_dtype=True}::test_binary
```
Comment (vlad-perevezentsev) [inline tests/skipped_tests_gpu_no_fp64.tbl:1257]: The same here

File: tests/skipped_tests_gpu_no_fp64.tbl [hunk 4]
```diff
@@ -930,1155 +930,249 @@ tests/third_party/cupy/manipulation_tests/test_basic.py::TestCopytoFromScalar_pa
...
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_184_{arg1=array([-3., -2., -1.,  1.,  2.,  3.]), arg2=-2.0, dtype=float64, name='true_divide', use_dtype=True}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_185_{arg1=array([-3., -2., -1.,  1.,  2.,  3.]), arg2=-2.0, dtype=float64, name='true_divide', use_dtype=False}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_186_{arg1=array([-3., -2., -1.,  1.,  2.,  3.]), arg2=-2.0, dtype=float64, name='subtract', use_dtype=True}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_187_{arg1=array([-3., -2., -1.,  1.,  2.,  3.]), arg2=-2.0, dtype=float64, name='subtract', use_dtype=False}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_188_{arg1=0.0, arg2=array([-3., -2., -1.,  1.,  2.,  3.], dtype=float32), dtype=float64, name='power', use_dtype=True}::test_binary
```
Comment (vlad-perevezentsev) [inline tests/skipped_tests_gpu_no_fp64.tbl:1275]: The same here

File: tests/skipped_tests_gpu_no_fp64.tbl [hunk 5]
```diff
@@ -930,1155 +930,249 @@ tests/third_party/cupy/manipulation_tests/test_basic.py::TestCopytoFromScalar_pa
...
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_196_{arg1=0.0, arg2=array([-3., -2., -1.,  1.,  2.,  3.]), dtype=float64, name='true_divide', use_dtype=True}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_197_{arg1=0.0, arg2=array([-3., -2., -1.,  1.,  2.,  3.]), dtype=float64, name='true_divide', use_dtype=False}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_198_{arg1=0.0, arg2=array([-3., -2., -1.,  1.,  2.,  3.]), dtype=float64, name='subtract', use_dtype=True}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_199_{arg1=0.0, arg2=array([-3., -2., -1.,  1.,  2.,  3.]), dtype=float64, name='subtract', use_dtype=False}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_218_{arg1=2.0, arg2=array([-3., -2., -1.,  1.,  2.,  3.], dtype=float32), dtype=float64, name='power', use_dtype=True}::test_binary
```
Comment (vlad-perevezentsev) [inline tests/skipped_tests_gpu_no_fp64.tbl:1284]: The same here

File: tests/skipped_tests_gpu_no_fp64.tbl [hunk 6]
```diff
@@ -930,1155 +930,249 @@ tests/third_party/cupy/manipulation_tests/test_basic.py::TestCopytoFromScalar_pa
...
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_226_{arg1=2.0, arg2=array([-3., -2., -1.,  1.,  2.,  3.]), dtype=float64, name='true_divide', use_dtype=True}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_227_{arg1=2.0, arg2=array([-3., -2., -1.,  1.,  2.,  3.]), dtype=float64, name='true_divide', use_dtype=False}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_228_{arg1=2.0, arg2=array([-3., -2., -1.,  1.,  2.,  3.]), dtype=float64, name='subtract', use_dtype=True}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_229_{arg1=2.0, arg2=array([-3., -2., -1.,  1.,  2.,  3.]), dtype=float64, name='subtract', use_dtype=False}::test_binary
-tests/third_party/cupy/math_tests/test_arithmetic.py::TestArithmeticBinary2_param_248_{arg1=-2.0, arg2=array([-3., -2., -1.,  1.,  2.,  3.], dtype=float32), dtype=float64, name='power', use_dtype=True}::test_binary
```
Comment (vlad-perevezentsev) [inline tests/skipped_tests_gpu_no_fp64.tbl:1293]: The same here

**Comment by vtavana** [comment]
These cases fail
FAILED tests/test_umath.py::test_umaths[('cos', 'd')0] - ValueError: Device Intel(R) Graphics [0x46a8] does not provide native support for double-precision floating point type.
FAILED tests/test_umath.py::test_umaths[('cos', 'd')1] - ValueError: Device Intel(R) Graphics [0x46a8] does not provide native support for double-precision floating point type.
FAILED tests/test_umath.py::test_umaths[('sin', 'd')0] - ValueError: Device Intel(R) Graphics [0x46a8] does not provide native support for double-precision floating point type.
FAILED tests/test_umath.py::test_umaths[('sin', 'd')1] - ValueError: Device Intel(R) Graphics [0x46a8] does not provide native support for double-precision floating point type.

Inside `skipped_tests_gpu_no_fp64.tbl` 
line 498 
- `tests/test_umath.py::test_umaths[('cos', 'd')]`
should be replaced with
- `tests/test_umath.py::test_umaths[('cos', 'd')0]`
- `tests/test_umath.py::test_umaths[('cos', 'd')1]`

and similarly line 503
- `tests/test_umath.py::test_umaths[('sin', 'd')]`

+ `tests/test_umath.py::test_umaths[('sin', 'd')0]`
+ `tests/test_umath.py::test_umaths[('sin', 'd')1]`

File: tests/test_umath.py [hunk 7]
```diff
@@ -88,8 +72,15 @@ def get_id(val):
 @pytest.mark.parametrize("test_cases", test_cases, ids=get_id)
```
Comment (vtavana) [inline tests/test_umath.py:72]: A few warnings are generated, you may want to add
`@pytest.mark.usefixtures("suppress_divide_invalid_numpy_warnings")`

tests/test_umath.py::test_umaths[('arccos', 'f')] /home/vtavanas/repos/dpnp/tests/test_umath.py:86: RuntimeWarning: invalid value encountered in arccos
  expected = getattr(numpy, umath)(*args)
PASSED
tests/test_umath.py::test_umaths[('arccosh', 'f')] /home/vtavanas/repos/dpnp/tests/test_umath.py:86: RuntimeWarning: invalid value encountered in arccosh
  expected = getattr(numpy, umath)(*args)
PASSED
tests/test_umath.py::test_umaths[('arcsin', 'f')] /home/vtavanas/repos/dpnp/tests/test_umath.py:86: RuntimeWarning: invalid value encountered in arcsin
  expected = getattr(numpy, umath)(*args)
PASSED
tests/test_umath.py::test_umaths[('arctanh', 'f')] /home/vtavanas/repos/dpnp/tests/test_umath.py:86: RuntimeWarning: divide by zero encountered in arctanh
  expected = getattr(numpy, umath)(*args)
/home/vtavanas/repos/dpnp/tests/test_umath.py:86: RuntimeWarning: invalid value encountered in arctanh
  expected = getattr(numpy, umath)(*args)
PASSED

Comment (antonwolfy) [reply tests/test_umath.py:73]: Yep, great idea. Thank you!

**Review by vtavana — APPROVED** [review]
Thank you, @antonwolfy

**Pull request merged**